src/main/kotlin/org/axonframework/intellij/ide/plugin/resolving/MessagePublisherResolver.kt

```diff
@@ -0,0 +1,51 @@
+package org.axonframework.intellij.ide.plugin.resolving
+
+import com.intellij.ide.highlighter.JavaFileType
+import com.intellij.openapi.project.Project
+import com.intellij.psi.JavaPsiFacade
+import com.intellij.psi.PsiElement
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.search.searches.MethodReferencesSearch
+import com.intellij.psi.util.CachedValue
+import com.intellij.psi.util.CachedValueProvider
+import com.intellij.psi.util.CachedValuesManager
+import com.intellij.psi.util.PsiModificationTracker
+import org.axonframework.intellij.ide.plugin.util.areAssignable
+import org.axonframework.intellij.ide.plugin.util.axonScope
+import org.axonframework.intellij.ide.plugin.util.comparePsiElementsBasedOnDisplayName
+import org.jetbrains.kotlin.idea.KotlinFileType
+import java.util.concurrent.ConcurrentHashMap
+
+class MessagePublisherResolver(val project: Project) {
```

Comment:
I might be wrong here, but this resolves the jump from message to handler, right? So, not from publisher to handler. 
My main doubt comes from the usage of "constructor publisher," which doesn't resonate directly with me.

Reply:
I've changed this to `MessageCreationResolver`, which describes the whole constructor-search somewhat better. Do you have another suggestion?